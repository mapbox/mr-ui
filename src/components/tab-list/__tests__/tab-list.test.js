import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { testCases } from './tab-list-test-cases';

describe('TabList', () => {
  let testCase;
  let wrapper;

  describe(testCases.basic.description, () => {
    beforeEach(() => {
      testCase = testCases.basic;
      wrapper = shallow(
        React.createElement(testCase.component, testCase.props)
      );
    });

    test('renders', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('onChange is called', () => {
      wrapper
        .find('button')
        .first()
        .props()
        .onClick('one');
      expect(testCase.props.onChange).toHaveBeenCalledTimes(1);
      expect(testCase.props.onChange).toHaveBeenCalledWith('one');
    });

    test('activeItem prop updates active item', () => {
      wrapper.setProps({ activeItem: 'one' });
      wrapper.update();
      expect(wrapper.find('.border--blue.txt-bold').length).toBe(1);
    });
  });
  describe(testCases.links.description, () => {
    beforeEach(() => {
      testCase = testCases.links;
      wrapper = shallow(
        React.createElement(testCase.component, testCase.props)
      );
    });

    test('renders', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('activeItem prop updates active item', () => {
      wrapper.setProps({ activeItem: 'one' });
      wrapper.update();
      expect(wrapper.find({ 'data-test': 'one' }).length).toBe(1);
    });
  });
  describe(testCases.truncateAll.description, () => {
    beforeEach(() => {
      testCase = testCases.truncateAll;
      wrapper = shallow(
        React.createElement(testCase.component, testCase.props)
      );
    });

    test('renders', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('activeItem prop updates active item', () => {
      wrapper.setProps({ activeItem: 'one' });
      wrapper.update();
      expect(wrapper.find({ 'data-test': 'one' }).length).toBe(1);
    });
  });
  describe(testCases.moreButtonOnlyIfNeeded.description, () => {
    beforeEach(() => {
      testCase = testCases.moreButtonOnlyIfNeeded;
      wrapper = shallow(
        React.createElement(testCase.component, testCase.props)
      );
    });

    test('renders', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('activeItem prop updates active item', () => {
      wrapper.setProps({ activeItem: 'one' });
      wrapper.update();
      expect(wrapper.find({ 'data-test': 'one' }).length).toBe(1);
    });
  });
  describe(testCases.labelNode.description, () => {
    beforeEach(() => {
      testCase = testCases.labelNode;
      wrapper = shallow(
        React.createElement(testCase.component, testCase.props)
      );
    });

    test('renders', () => {
      expect(toJson(wrapper)).toMatchSnapshot();
    });

    test('onChange is called', () => {
      wrapper
        .find('button')
        .first()
        .props()
        .onClick('one');
      expect(testCase.props.onChange).toHaveBeenCalledTimes(1);
      expect(testCase.props.onChange).toHaveBeenCalledWith('one');
    });

    test('activeItem prop updates active item', () => {
      wrapper.setProps({ activeItem: 'one' });
      wrapper.update();
      expect(wrapper.find({ 'data-test': 'one' }).length).toBe(1);
    });
  });
});
