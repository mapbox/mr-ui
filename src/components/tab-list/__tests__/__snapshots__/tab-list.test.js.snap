// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabList a label contains a node renders 1`] = `
<div
  className="clearfix"
>
  <div
    className="fl"
    key="0"
  >
    <button
      aria-label="one"
      className="px6 mr24 fl-mm align-center-mm align-l py12 border-b-mm border-b--2 inline-block border-b border--transparent "
      data-test="one"
      key="one"
      onClick={[Function]}
      type="button"
    >
      <Icon
        inline={false}
        name="star"
        passthroughProps={Object {}}
        size={18}
      />
    </button>
  </div>
  <div
    className="fl"
    key="1"
  >
    <button
      aria-label="two"
      className="px6 mr24 fl-mm align-center-mm align-l py12 border-b-mm border-b--2 inline-block border-b border--transparent "
      data-test="two"
      key="two"
      onClick={[Function]}
      type="button"
    >
      Label two
    </button>
  </div>
  <div
    className="fl"
    key="0"
  >
    <button
      aria-label="three"
      className="px6 mr24 fl-mm align-center-mm align-l py12 border-b-mm border-b--2 inline-block-mm none border--blue txt-bold "
      data-test="three"
      key="three"
      onClick={[Function]}
      type="button"
    >
      Label three
    </button>
  </div>
  <div
    className="fl"
    key="1"
  >
    <button
      aria-label="four"
      className="px6 mr24 fl-mm align-center-mm align-l py12 border-b-mm border-b--2 inline-block-mm none border--transparent "
      data-test="four"
      key="four"
      onClick={[Function]}
      type="button"
    >
      Label four
    </button>
  </div>
  <div
    className="fl"
  >
    <PopoverTrigger
      block={false}
      content={
        Array [
          <div
            className="block"
          >
            <button
              aria-label="three"
              className="px6 mr24 fl-mm align-center-mm align-l py12 border-b-mm border-b--2 border-b--0 border--blue txt-bold "
              data-test="three"
              onClick={[Function]}
              type="button"
            >
              Label three
            </button>
          </div>,
          <div
            className="block"
          >
            <button
              aria-label="four"
              className="px6 mr24 fl-mm align-center-mm align-l py12 border-b-mm border-b--2 border-b--0 border--transparent "
              data-test="four"
              onClick={[Function]}
              type="button"
            >
              Label four
            </button>
          </div>,
        ]
      }
      disabled={false}
      onPopoverClose={[Function]}
      onPopoverOpen={[Function]}
      popoverProps={
        Object {
          "placement": "bottom",
          "themePopover": "round shadow-darken25 h480 scroll-auto px12 py12 scroll-styled",
        }
      }
      receiveFocus={true}
      renderHiddenContent={false}
      respondsToClick={true}
      respondsToFocus={false}
      respondsToHover={true}
      trapFocus={true}
      triggerProps={
        Object {
          "data-test": "more-dropdown-menu",
        }
      }
    >
      <button
        aria-label="More"
        className="px6 py12 mr12 align-l cursor-pointer none-mm"
        type="button"
      >
        More +
      </button>
    </PopoverTrigger>
  </div>
</div>
`;

exports[`TabList basic renders 1`] = `
<div
  className="clearfix"
>
  <div
    className="fl"
    key="0"
  >
    <button
      aria-label="one"
      className="px6 mr24 fl-mm align-center-mm align-l py12 border-b-mm border-b--2 inline-block border-b border--transparent "
      data-test="one"
      key="one"
      onClick={[Function]}
      type="button"
    >
      Label one
    </button>
  </div>
  <div
    className="fl"
    key="1"
  >
    <button
      aria-label="two"
      className="px6 mr24 fl-mm align-center-mm align-l py12 border-b-mm border-b--2 inline-block border-b border--transparent "
      data-test="two"
      key="two"
      onClick={[Function]}
      type="button"
    >
      Label two
    </button>
  </div>
  <div
    className="fl"
    key="0"
  >
    <button
      aria-label="three"
      className="px6 mr24 fl-mm align-center-mm align-l py12 border-b-mm border-b--2 inline-block-mm none border--blue txt-bold "
      data-test="three"
      key="three"
      onClick={[Function]}
      type="button"
    >
      Label three
    </button>
  </div>
  <div
    className="fl"
    key="1"
  >
    <button
      aria-label="four"
      className="px6 mr24 fl-mm align-center-mm align-l py12 border-b-mm border-b--2 inline-block-mm none border--transparent "
      data-test="four"
      key="four"
      onClick={[Function]}
      type="button"
    >
      Label four
    </button>
  </div>
  <div
    className="fl"
  >
    <PopoverTrigger
      block={false}
      content={
        Array [
          <div
            className="block"
          >
            <button
              aria-label="three"
              className="px6 mr24 fl-mm align-center-mm align-l py12 border-b-mm border-b--2 border-b--0 border--blue txt-bold "
              data-test="three"
              onClick={[Function]}
              type="button"
            >
              Label three
            </button>
          </div>,
          <div
            className="block"
          >
            <button
              aria-label="four"
              className="px6 mr24 fl-mm align-center-mm align-l py12 border-b-mm border-b--2 border-b--0 border--transparent "
              data-test="four"
              onClick={[Function]}
              type="button"
            >
              Label four
            </button>
          </div>,
        ]
      }
      disabled={false}
      onPopoverClose={[Function]}
      onPopoverOpen={[Function]}
      popoverProps={
        Object {
          "placement": "bottom",
          "themePopover": "round shadow-darken25 h480 scroll-auto px12 py12 scroll-styled",
        }
      }
      receiveFocus={true}
      renderHiddenContent={false}
      respondsToClick={true}
      respondsToFocus={false}
      respondsToHover={true}
      trapFocus={true}
      triggerProps={
        Object {
          "data-test": "more-dropdown-menu",
        }
      }
    >
      <button
        aria-label="More"
        className="px6 py12 mr12 align-l cursor-pointer none-mm"
        type="button"
      >
        More +
      </button>
    </PopoverTrigger>
  </div>
</div>
`;

exports[`TabList links renders 1`] = `
<div
  className="clearfix"
>
  <div
    className="fl"
    key="0"
  >
    <a
      className="inline-block px6 mr24 fl-mm align-center-mm align-l py12 border-b-mm border-b--2 inline-block border-b border--blue txt-bold "
      data-test="one"
      href="#"
      key="one"
    >
      Label one
    </a>
  </div>
  <div
    className="fl"
    key="1"
  >
    <a
      className="inline-block px6 mr24 fl-mm align-center-mm align-l py12 border-b-mm border-b--2 inline-block border-b border--transparent "
      data-test="two"
      href="#"
      key="two"
    >
      Label two
    </a>
  </div>
  <div
    className="fl"
    key="0"
  >
    <a
      className="inline-block px6 mr24 fl-mm align-center-mm align-l py12 border-b-mm border-b--2 inline-block-mm none border--transparent "
      data-test="three"
      href="#"
      key="three"
    >
      Label three
    </a>
  </div>
  <div
    className="fl"
    key="1"
  >
    <a
      className="inline-block px6 mr24 fl-mm align-center-mm align-l py12 border-b-mm border-b--2 inline-block-mm none border--transparent "
      data-test="four"
      href="#"
      key="four"
    >
      Label four
    </a>
  </div>
  <div
    className="fl"
  >
    <PopoverTrigger
      block={false}
      content={
        Array [
          <div
            className="block"
          >
            <a
              className="inline-block px6 mr24 fl-mm align-center-mm align-l py12 border-b-mm border-b--2 border-b--0 border--transparent "
              data-test="three"
              href="#"
            >
              Label three
            </a>
          </div>,
          <div
            className="block"
          >
            <a
              className="inline-block px6 mr24 fl-mm align-center-mm align-l py12 border-b-mm border-b--2 border-b--0 border--transparent "
              data-test="four"
              href="#"
            >
              Label four
            </a>
          </div>,
        ]
      }
      disabled={false}
      onPopoverClose={[Function]}
      onPopoverOpen={[Function]}
      popoverProps={
        Object {
          "placement": "bottom",
          "themePopover": "round shadow-darken25 h480 scroll-auto px12 py12 scroll-styled",
        }
      }
      receiveFocus={true}
      renderHiddenContent={false}
      respondsToClick={true}
      respondsToFocus={false}
      respondsToHover={true}
      trapFocus={true}
      triggerProps={
        Object {
          "data-test": "more-dropdown-menu",
        }
      }
    >
      <button
        aria-label="More"
        className="px6 py12 mr12 align-l cursor-pointer none-mm"
        type="button"
      >
        More +
      </button>
    </PopoverTrigger>
  </div>
</div>
`;

exports[`TabList truncate all renders 1`] = `
<div
  className="clearfix"
>
  <div
    className="fl"
    key="0"
  >
    <a
      className="inline-block px6 mr24 fl-mm align-center-mm align-l py12 border-b-mm border-b--2 inline-block-mm none border--blue txt-bold "
      data-test="one"
      href="#"
      key="one"
    >
      Label one
    </a>
  </div>
  <div
    className="fl"
    key="1"
  >
    <a
      className="inline-block px6 mr24 fl-mm align-center-mm align-l py12 border-b-mm border-b--2 inline-block-mm none border--transparent "
      data-test="two"
      href="#"
      key="two"
    >
      Label two
    </a>
  </div>
  <div
    className="fl"
    key="2"
  >
    <a
      className="inline-block px6 mr24 fl-mm align-center-mm align-l py12 border-b-mm border-b--2 inline-block-mm none border--transparent "
      data-test="three"
      href="#"
      key="three"
    >
      Label three
    </a>
  </div>
  <div
    className="fl"
    key="3"
  >
    <a
      className="inline-block px6 mr24 fl-mm align-center-mm align-l py12 border-b-mm border-b--2 inline-block-mm none border--transparent "
      data-test="four"
      href="#"
      key="four"
    >
      Label four
    </a>
  </div>
  <div
    className="fl"
  >
    <PopoverTrigger
      block={false}
      content={
        Array [
          <div
            className="block"
          >
            <a
              className="inline-block px6 mr24 fl-mm align-center-mm align-l py12 border-b-mm border-b--2 border-b--0 border--blue txt-bold "
              data-test="one"
              href="#"
            >
              Label one
            </a>
          </div>,
          <div
            className="block"
          >
            <a
              className="inline-block px6 mr24 fl-mm align-center-mm align-l py12 border-b-mm border-b--2 border-b--0 border--transparent "
              data-test="two"
              href="#"
            >
              Label two
            </a>
          </div>,
          <div
            className="block"
          >
            <a
              className="inline-block px6 mr24 fl-mm align-center-mm align-l py12 border-b-mm border-b--2 border-b--0 border--transparent "
              data-test="three"
              href="#"
            >
              Label three
            </a>
          </div>,
          <div
            className="block"
          >
            <a
              className="inline-block px6 mr24 fl-mm align-center-mm align-l py12 border-b-mm border-b--2 border-b--0 border--transparent "
              data-test="four"
              href="#"
            >
              Label four
            </a>
          </div>,
        ]
      }
      disabled={false}
      onPopoverClose={[Function]}
      onPopoverOpen={[Function]}
      popoverProps={
        Object {
          "placement": "bottom",
          "themePopover": "round shadow-darken25 h480 scroll-auto px12 py12 scroll-styled",
        }
      }
      receiveFocus={true}
      renderHiddenContent={false}
      respondsToClick={true}
      respondsToFocus={false}
      respondsToHover={true}
      trapFocus={true}
      triggerProps={
        Object {
          "data-test": "more-dropdown-menu",
        }
      }
    >
      <button
        aria-label="More"
        className="px6 py12 mr12 align-l cursor-pointer none-mm"
        type="button"
      >
        More +
      </button>
    </PopoverTrigger>
  </div>
</div>
`;
